packages/babel-plugin/src/class-names/__tests__/object-literal.test.tsx

```diff
@@ -128,4 +128,96 @@ describe('class names object literal', () => {
 
     expect(actual).toIncludeMultiple([':hover{color:red}', '{font-size:20px}']);
   });
+
+  it('should transform object with argument arrow function variable', () => {
+    const actual = transform(`
+        import { ClassNames } from '@compiled/react';
+
+        const color1 = 'black';
+        const mixin = ({ color1, color2: c }, color3, radius) => ({
+          color: color1,
+          backgroundColor: c,
+          borderColor: color3 ,
+          borderRadius: radius,
+        });
+
+        const color = { red: 'red' };
+        const greenColor = 'green';
+
+        const Component = (props) => {
+          const color2 = 'black';
+
+          return (
+            <ClassNames>
+              {({ css }) => <div className={css({ ...mixin({ color1: color.red, color2: 'blue' }, greenColor, 10) })} />}
+            </ClassNames>
+          );
+        };
+    `);
+
+    expect(actual).toIncludeMultiple([
+      '{color:red}',
+      '{background-color:blue}',
+      '{border-color:green}',
+      '{border-radius:10px}',
+    ]);
+  });
+
+  it('should transform object with unresolved argument arrow function variable', () => {
+    const actual = transform(`
+        import { ClassNames } from '@compiled/react';
+
+        const radius = 10;
+        const mixin = (color1, radius, size, weight) => ({
+          color: color1,
+          borderRadius: radius,
+          fontSize: size,
+          fontWeight: weight
+        });
+
+        const Component = (props) => (
+          <ClassNames>
+            {({ css, style }) => <div style={style} className={css({ ...mixin(props.color1, radius) })} />}
+          </ClassNames>
+        );
+      `);
+
+    expect(actual).toIncludeMultiple([
+      '{color:var(--_zo7lop)}',
+      '"--_zo7lop":ix(props.color1)',
+      '{border-radius:10px}',
+      '{font-weight:var(--_u6vle4)}',
+      '"--_u6vle4":ix()',
```

Comment:
this doesn't look right

Reply:
🤔 So this is the case when passed args and params count doesn't matched. When we passed less args than params. 🤔 Should we remove this altogether?

Reply:
ah yeah i see now.

hmm, i think it could go both ways. what do you think

Reply:
🤔 Let me see if its possible to remove `'"--_u6vle4":ix()'` completely if not param is not resolved.

Reply:
Tried. Need lot of code change for this 😞 Can we create a card for this cleanup?

Reply:
i don't think its the end of the world, let's keep it for now.